includes/admin/class-wcmp-api.php

```diff
@@ -159,13 +162,32 @@ public function getShipmentLabels(array $shipment_ids, array $order_ids, array $
 
         if ($display) {
             $collection->setPdfOfLabels($positions);
-            WCMP_Export::saveTrackTracesToOrders($collection, $order_ids);
+            $this->updateOrderBarcode($order_ids, $collection);
             $collection->downloadPdfOfLabels($display);
-        } else {
+        }
+
+        if (! $display) {
             $collection->setLinkOfLabels($positions);
-            WCMP_Export::saveTrackTracesToOrders($collection, $order_ids);
+            $this->updateOrderBarcode($order_ids, $collection);
             echo $collection->getLinkOfLabels();
-            die();
         }
     }
-}
+
+    /**
+     * @param array $orderIds
+     * @param MyParcelCollection $collection
+     * @throws Exception
+     */
+    private function updateOrderBarcode(array $orderIds, MyParcelCollection $collection) : void
+    {
+        foreach ($orderIds as $orderId) {
+            $order = WC_Core::get_order($orderId);
+            $lastShipmentIds = unserialize($order->get_meta('_myparcel_last_shipment_ids'));
+            $shipmentData = (new WCMP_Export())->getShipmentData($lastShipmentIds, $order);
+            $trackTrace = $shipmentData["track_trace"];
+            ChannelEngine::updateMetaOnExport($order, $trackTrace);
+        }
+
+        WCMP_Export::saveTrackTracesToOrders($collection, $orderIds);
+    }
+}
```

Comment:
newline, stel ff je editor goed in dat hij dat voor je doet desnoods

Reply:
Heb editor ingesteld, en new line toegevoegd